guide/popular-topics/errors.md

```diff
@@ -0,0 +1,155 @@
+# Errors
+
+There is no doubt that you have encountered errors while making bots. While errors are very useful at warning you of what is going wrong, many people are stumped by them and how to track them down and fix them, but don't worry, we have you covered. This section will be all about how to diagnose errors, how to identify where they are coming from, and how to fix them.
+
+## Types of Errors
+
+### API Errors
+
+API Errors or DiscordAPIErrors are errors which are thrown by the Discord API when an invalid request is carried out. API Errors can be mostly diagnosed using the message that is given. They can also be further examined by seeing the http method and path used, we will explore tracking these errors down in the next section.
+
+Example: `DiscordAPIError: Cannot send an empty message`
+
+### Discord.js Errors
+
+Discord.js Errors are errors which are thrown by the library itself, they can usually be easily tracked down using the stacktrace and error message.
+
+Example: `Error: Members didn't arrive in time.`
+
+### JS Errors
+
+JS Errors are simple errors which can be thrown by node itself, or by discord.js. These types of errors can easily be fixed by looking at the type of error, and the stacktrace. You can find a full list of types [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error) And a list of common js errors [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors).
+
+Example: `ReferenceError: "x" is not defined`
+
+### Websocket and Network Errors
+
+Websocket and Network errors are common system errors thrown by Node in response to something wrong with the websocket connection. Unfortunately, these errors do not have a concrete solution and can be (usually) fixed by getting a better, and stronger connection. However, if you handle these errors correctly, discord.js will attempt to reconnect to the websocket automatically.
+
+You can handle these errors by adding an error listener as shown below.
+
+```js
+client.on('error', console.error);
+```
+
+Now, when an error occurs it will be logged to the console and discord.js will automatically attempt to reconnect to the websocket.
```

Comment:
As said above, at least in master, you don't need to log them, d.js will still attempt to reconnect you

Reply:
yep, will refactor all of this for master